### DISCUSSION TITLE
Custom extensions, access to previous messages?

### DISCUSSION THREAD

**Discussion by christian-nils** [foxglove/community#850] [Q&A] [answered]
Hi,

I am trying to modify https://github.com/codewithpassion/foxglove-studio-h264-extension to be able to replay h264 video data stored in a (ROS2) MCAP file. It works quite well now, except when seeking through the video there are artifacts until an IDR frame is reached. As explained in other discussions, it would be nice to access messages preceding the currentFrame just after seeking.

When using allFrames, it somewhat works but has a lot of limitations. The one I experienced is when preloading the topic, the cache gets full and obviously allFrames don't contain all frames.

Is there any other ways of getting frames preceding the currentFrame? If not, do you plan to have a similar mechanism on your roadmap? Or implement an official compressed video player panel?

Thanks in advance,

**Comment by jtbandes** [comment, marked as answer]
Yes, official compressed video support is in the works and will be announced soon.

**Reply by christian-nils** [reply]
Awesome! Thanks for your reply, nice to hear that this will be soon announced. Have a great day.
